nixos/modules/services/continuous-integration/buildbot-master/default.nix

```diff
@@ -0,0 +1,243 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+  cfg = config.services.buildbot-master;
+  configFile = pkgs.writeText "master.cfg" ''
+
+from buildbot.plugins import *
+c = BuildmasterConfig = {}
+c['workers'] = [ worker.Worker('${cfg.workerUser}', '${cfg.workerPass}') ]
+c['protocols'] = {'pb': {'port': ${cfg.workerPort}}}
+c['title'] = "${cfg.title}"
+c['titleURL'] = "${cfg.titleUrl}"
+c['buildbotURL'] = "${cfg.buildbotUrl}"
+c['db'] = { 'db_url' : "${cfg.dbUrl}" }
+c['www'] = dict(port=${cfg.port})
+
+factory = util.BuildFactory()
+for step in [ ${cfg.factorySteps} ]:
+  factory.addStep(step)
+
+c['change_source'] = []
+for source in [ ${cfg.changeSource} ]:
+  c['change_source'].append(source)
+
+c['schedulers'] = []
+for sched in [ ${cfg.schedulers} ]:
+  c['schedulers'].append(sched)
+
+c['builders'] = []
+for build in [ ${cfg.builders} ]:
+  c['builders'].append(build)
+
+c['status'] = []
+for stat in [ ${cfg.status} ]:
+  c['status'].append(stat)
+'';
+
+in {
+  options = {
+    services.buildbot-master = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable the buildbot continuous integration server.";
+      };
+
+      factorySteps = mkOption {
+        default = "\\
+          \"steps.Git(repourl='git://github.com/buildbot/pyflakes.git', mode='incremental')\", \\
+          \"steps.ShellCommand(command=['trial', 'pyflakes'])\" \\
+        ";
+        type = types.str;
+        description = "Factory Steps";
+      };
+
+      workerUser = mkOption {
+        default = "example-worker";
+        type = types.str;
+        description = "Buildbot Worker User.";
+      };
+
+      workerPass = mkOption {
+        default = "pass";
+        type = types.str;
+        description = "Buildbot Worker Password.";
+      };
+
+      workerPort = mkOption {
+        default = "9989";
+        type = types.str;
+        description = "Buildbot Worker Port.";
+      };
+
+      changeSource = mkOption {
+        default = " \\
+          \"changes.GitPoller('git://github.com/buildbot/pyflakes.git', workdir='gitpoller-workdir', branch='master', pollinterval=300)\" \\
```

Comment:
These defaults seem more appropriate for `example`, than `default`. `example` just shows up in the documentation as possible values, whereas `default` will actually get set if you don't set some other value for it. Given that most people probably don't need to monitor the pyflakes repo, it makes more sense to me for the config to complain that this hasn't been set than to default to looking somewhere many users might not want to look.